handshake/stk_generator.go

```diff
@@ -0,0 +1,72 @@
+package handshake
+
+import (
+	"bytes"
+	"net"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/crypto"
+)
+
+// An STK is a source address token
+type STK struct {
+	RemoteAddr string
+	SentTime   time.Time
+}
+
+// An STKGenerator generates STKs
+type STKGenerator struct {
```

Comment:
I don't think we can use Go's binary encoding, because the data in the source address has variable length. I tried it in the playground: https://play.golang.org/p/3j_VzfKxa-.